sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/data/InvitationsCollectionAttributes.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.benchmark.linkedin.data;
+
+import com.azure.cosmos.benchmark.linkedin.impl.Constants;
+import com.azure.cosmos.models.ExcludedPath;
+import com.azure.cosmos.models.IncludedPath;
+import com.azure.cosmos.models.IndexingMode;
+import com.azure.cosmos.models.IndexingPolicy;
+import com.google.common.collect.ImmutableList;
+
+
+/**
+ * Encapsulates the Indexing policy specific to the Invitations collections
+ */
+class InvitationsCollectionAttributes implements CollectionAttributes {
+    private static final IncludedPath INCLUDED_PATH = new IncludedPath(Constants.PARTITIONING_KEY_INDEXING_INCLUDE_PATH);
+    private static final ExcludedPath EXCLUDED_PATH = new ExcludedPath(Constants.WILDCARD_INDEXING_EXCLUDE_PATH);
+
+    private final IndexingPolicy _indexingPolicy;
+
+    InvitationsCollectionAttributes() {
+        _indexingPolicy = new IndexingPolicy()
+            .setAutomatic(true)
+            .setIncludedPaths(ImmutableList.of(INCLUDED_PATH))
```

Comment:
Why we want to include partition key path, is it because we want the query on partition. We can write query by passing partition key in the option instead of query itself , no ? Also For bulk for bulk insert we can remove all indexing and then introduce again for regular work flow.

Reply:
Yes, that's why partitioningKey is explicitly included. We are already passing the partitioningKey in the option for the query. My understanding of the IndexingDirective in CosmosItemRequestOptions was to selectively include/exclude indexing for a record at creation/update time. If a consistent policy needs to be applied for all records, the collection's indexingPolicy is where you define it (so one doesn't have the added burden of including the IndexingDirective in every call)

The disable indexing, then reenable indexing can be done. Given 2 columns are indexed (partitioningKey, and id by default), it's going to complicate the implementation for a marginal benefit?

Reply:
Based on our offline conversation, we will validate correct partitioningKey specification in the queryItems change